embeddings/pipeline/sklearn_classification.py

```diff
@@ -0,0 +1,190 @@
+        self._prepare_data()
+        self.random_forest = None
+        self.tree = None
+        self.logistic = None
+        self.vectorizer = None
+
+    def _prepare_data(self) -> None:
+
+        dataset = datasets.load_dataset(self.dataset_name)
+
+        for row in dataset["train"]:
+            self.df_train = self.df_train.append(row, ignore_index=True)
+
+        for row in dataset["validation"]:
+            self.df_valid = self.df_valid.append(row, ignore_index=True)
+
+        for row in dataset["test"]:
+            self.df_test = self.df_test.append(row, ignore_index=True)
+
+    def vectorize_text(
+        self, vectorization_method: str = "tfidf"
+    ) -> Tuple[np.ndarray, np.ndarray, np.ndarray]:
+
+        if vectorization_method == "bow":
+            self.vectorizer = CountVectorizer(max_features=self.max_features)
+        elif vectorization_method == "tfidf":
+            self.vectorizer = TfidfVectorizer(max_features=self.max_features)
+
+        self.vectorizer.fit(self.df_train[self.feature_name])
+
+        train_data = self.vectorizer.transform(self.df_train[self.feature_name]).toarray()
+        valid_data = self.vectorizer.transform(self.df_valid[self.feature_name]).toarray()
+        test_data = self.vectorizer.transform(self.df_test[self.feature_name]).toarray()
+
+        return train_data, valid_data, test_data
+
+    def train_decision_tree(self, train_data):
+
+        target = self.df_train[self.target_name].values
+        tree = DecisionTreeClassifier()
+        tree.fit(train_data, target)
+        self.tree = tree
+
+    def train_logistic(self, train_data):
+
+        target = self.df_train[self.target_name].values
+        logistic = LogisticRegression()
+        logistic.fit(train_data, target)
+        self.logistic = logistic
+
+    def train_random_forest(self, train_data):
+
+        target = self.df_train[self.target_name].values
+        random_forest = RandomForestClassifier()
+        random_forest.fit(train_data, target)
+        self.random_forest = random_forest
+
+    def evaluate_models(
+        self, train_data, valid_data, test_data
+    ) -> Dict[str, Dict[str, Dict[str, float]]]:
+
+        target_train = self.df_train[self.target_name].values
+        target_valid = self.df_valid[self.target_name].values
+        target_test = self.df_test[self.target_name].values
+
+        tree_preds_train = self.tree.predict(train_data)
+        tree_preds_valid = self.tree.predict(valid_data)
+        tree_preds_test = self.tree.predict(test_data)
+
+        logistic_preds_train = self.logistic.predict(train_data)
+        logistic_preds_valid = self.logistic.predict(valid_data)
+        logistic_preds_test = self.logistic.predict(test_data)
+
+        random_forest_preds_train = self.random_forest.predict(train_data)
+        random_forest_preds_valid = self.random_forest.predict(valid_data)
+        random_forest_preds_test = self.random_forest.predict(test_data)
+
+        return {
```

Comment:
Initial Version Remark: We have evaluator for such purpose [TextClassificationEvaluator](https://github.com/CLARIN-PL/embeddings/tree/main/embeddings/evaluator/text_classification_evaluator.py#L11)